packages/components/src/custom-select-control-v2/legacy-component/test/index.tsx

```diff
@@ -53,7 +62,9 @@ const ControlledCustomSelectControl = ( {
 	onChange,
 	...restProps
 }: React.ComponentProps< typeof UncontrolledCustomSelectControl > ) => {
+	const { value: overrideValue } = restProps;
 	const [ value, setValue ] = useState( options[ 0 ] );
+	const initialValue = overrideValue ?? value;
```

Comment:
Could we achieve the same behavior by passing `props.value` to the `useState` hook ? E.g.

```
const [ value, setValue ] = useState( restProps.value ?? options[ 0 ] );
```